tests/goal/test_goal.py

```diff
@@ -80,6 +80,30 @@ def test_goal_simple_args() -> None:
     verify(result.output)
 
 
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_input_file() -> None:
```

Comment:
aside from this, make sure to check on other goal command that may require dealing with files, if you see any interesting edge cases or alternative uses (for instance whether there are any command that allow for N input and N outputs) make sure to cover such edge cases too.